src/pages/Homepage/index.jsx

```diff
@@ -1,35 +1,27 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import { Route, Link } from "react-router-dom";
 import { API_KEY, URL } from "../../config.js";
 import "./homepage.styles.scss";
 import Spinner from "../../component/Spinner";
 import GifCard from "../../component/GifCard";
 import GifDetails from "../Details/index.jsx";
 
-class Homepage extends Component {
-  constructor(props) {
-    super(props);
+const Homepage = (props) => {
+  const [gifsData, setGifsData] = useState([]);
+  const [keywords, setKeywords] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
 
-    this.state = {
-      gifsData: [],
-      keywords: "",
-      isLoading: false,
-      errorMessage: "",
-    };
-  }
-
-  handleChange = (event) => {
-    const { name, value } = event.target;
-    this.setState({ [name]: value });
+  const handleChange = (event) => {
+    const { value } = event.target;
+    setKeywords(value);
   };
 
-  handleSubmit = (event) => {
-    const { keywords } = this.state;
-
+  const handleSubmit = (event) => {
     event.preventDefault();
     keywords === ""
-      ? this.setState({ errorMessage: "Enter a keyword" })
-      : this.setState({ isLoading: true, errorMessage: "" });
+      ? setErrorMessage("Enter a keyword")
+      : setIsLoading(true) && setErrorMessage("");
 
     fetch(`${URL}${API_KEY}&q=${keywords}&limit=30&offset=0&rating=G&lang=en`, {
```

Comment:
Nice use of string building. An idea would be to put the query parameters e.g `limit, offset, rating and lang` in the config. Not a big deal for this app but for bigger apps its good practice!